gdbstub_arch/src/msp430/mod.rs

```diff
@@ -9,18 +9,38 @@ pub mod reg;
 ///
 /// Check out the [module level docs](gdbstub::arch#whats-with-regidimpl) for
 /// more info about the `RegIdImpl` type parameter.
-pub enum Msp430<RegIdImpl: RegId = reg::id::Msp430RegId> {
+pub enum Msp430<RegIdImpl: RegId = reg::id::Msp430RegId<u16>> {
```

Comment:
While you're here, can you remove the `RegIdImpl` parameter, and use a fixed `reg::id::Msp430RegId<u16>` instead?